packages/http-server-javascript/src/common/serialization/json.ts

```diff
@@ -276,6 +324,47 @@ function transposeExpressionToJson(
   }
 }
 
+function getScalarEncoder(ctx: SerializationContext, type: Scalar, scalar: JsScalar) {
+  const encoding = getEncode(ctx.program, type);
+
+  let encoder: Encoder;
+
+  if (encoding) {
+    const encodingName = encoding.encoding ?? "default";
+    const scalarEncoder = scalar.getEncoding(encodingName, encoding.type);
+
+    // TODO/witemple - we should detect this before realizing models and use a transform to
```

Comment:
I like having issue references with TODO comments, is there an issue we can correlate with this?

Reply:
I think the most logical place for us to handle this is in the same codepath that we handle visibility transforms. As we're walking the models with MetadataInfo to determine whether and how the model needs to be transformed, we can also check required encodings on properties and scalars to determine if "decay" is required. So a model `Foo` could become `FooCreateItemJson` as we're walking it and determining that (1) the visibility Create | Item cannot be shared with the canonical representation and (2) it has a field that we cannot encode, so the property "decays" to a JSON form.

I'll add a note about this to the visibility transform issue, as this is relatively simple to detect while we're walking the type for that purpose I think.